UI/EncyclopediaDetailPanel.cpp

```diff
@@ -987,6 +1006,25 @@ void EncyclopediaDetailPanel::HandleSearchTextEntered() {
         }
     }
 
+    // search for matches within article text
+    match_report += "\n" + UserString("ENC_SEARCH_ARTICLE_MATCHES") + "\n\n";
+    for (std::multimap<std::string, std::pair<std::string, std::string> >::const_iterator
+         entry_it = all_pedia_entries_list.begin();
+         entry_it != all_pedia_entries_list.end(); ++entry_it)
+    {
+        std::set<std::pair<std::string, std::string> >::const_iterator dupe_it =
+            already_listed_results.find(std::make_pair(entry_it->second.second, entry_it->first));
+        if (dupe_it != already_listed_results.end())
+            continue;
+
+        EncyclopediaArticle article_entry = GetPediaArticle(entry_it->second.second);
+        if (boost::icontains(UserString(article_entry.description), search_text)) {
```

Comment:
Added config option.  Instead of adding another selection in the options screen, I'd like to defer the UI counterpart for #755